WalletWasabi.Fluent/ViewModels/Wallets/Home/History/Details/TransactionDetailsViewModel.cs

```diff
@@ -46,19 +46,18 @@ public TransactionDetailsViewModel(UiContext uiContext, IWalletModel wallet, Tra
 
 	public uint256 TransactionId { get; }
 
-	public Amount? Fee { get; }
-
 	public ICollection<BitcoinAddress> DestinationAddresses { get; }
 
-	public bool IsFeeVisible { get; }
-
 	private void UpdateValues(TransactionModel model)
 	{
 		DateString = model.DateString;
 		Labels = model.Labels;
 		BlockHeight = model.BlockHeight;
 		Confirmations = model.Confirmations;
 
+		Fee = UiContext.AmountProvider.Create(model.Fee());
+		IsFeeVisible = Fee != null && Fee.HasBalance;
```

Comment:
Fee is never null here

(Out of scope but the naming of `HasBalance` is confusing)

Reply:
Fee can be and is null when the tx has any Foreign input. That's why we need the help of the `TransactionFeeProvider`.

I didn't name `HasBalance`, I only use it.

Reply:
`model.Fee()` can be null, `Fee` can't

Reply:
OK, i will just remove the Fee != null check. It was just a sanity check or a "just to be sure" check.